src/fparser/two/Fortran2003.py

```diff
@@ -10157,6 +10159,7 @@ def match(reader):
             [Specification_Part, Module_Subprogram_Part],
             End_Module_Stmt,
             reader,
```

Comment:
Each sub-class can only match once for a valid module? (I initially got worried that this wouldn't work for multiple modules in a file but it's fine.)

Reply:
Yes, a Module can only contain one Module_Stmt, one Specification_Part, one Module_Subprogram_Part and one End_Module_Stmt. Multiple modules are supported in 'higher level' rules, such as Program_Unit.